astropy/utils/parsing.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Wrappers for PLY to provide thread safety.
+"""
+
+_TAB_HEADER = """# -*- coding: utf-8 -*-
```

Comment:
Should `_add_tab_header` be public too?

Reply:
I felt it was a bit overly specific, but I guess if the header is public, the routine that adds it might also be. But maybe not - functions do get typeset while the variables do not (by default).

Reply:
Note that TAB_HEADER needs to be passed through `str.format` to insert the relevant package name. My concern with making TAB_HEADER public rather than add_tab_header is that if we ever decide to add another field to format (e.g. if for some reason we chose to insert a timestamp, not that I'm saying that would be a good idea), code that uses TAB_HEADER directly would probably break.

Reply:
OK, good point, let's keep it private.